src/api/kumaApi.ts

```diff
@@ -1,227 +1,2 @@
-    return this.client.get(`meshes/${mesh}/dataplanes/${name}/policies`, { params })
-  }
-
-  getDataplaneRules({ mesh, name }: { mesh: string; name: string }, params?: any): Promise<ApiListResponse<DataplaneRule>> {
-    return this.client.get(`meshes/${mesh}/dataplanes/${name}/rules`, { params })
-  }
-
-  /**
-   * Fetches additional data like xDS configuration, envoy stats, or envoy clusters.
-   */
-  getDataplaneData({ mesh, dppName, dataPath }: { mesh: string, dppName: string, dataPath: 'xds' | 'stats' | 'clusters' }, params?: any): Promise<string> {
-    return this.client.get(`meshes/${mesh}/dataplanes/${dppName}/${dataPath}`, { params })
-  }
-
-  getAllServiceInsights(params?: PaginationParameters): Promise<PaginatedApiListResponse<ServiceInsight>> {
-    return this.client.get('service-insights', { params })
-  }
-
-  getAllServiceInsightsFromMesh({ mesh }: { mesh: string }, params?: PaginationParameters): Promise<PaginatedApiListResponse<ServiceInsight>> {
-    return this.client.get(`meshes/${mesh}/service-insights`, { params })
-  }
-
-  getServiceInsight({ mesh, name }: { mesh: string, name: string }, params?: any): Promise<ServiceInsight> {
-    return this.client.get(`meshes/${mesh}/service-insights/${name}`, { params })
-  }
-
-  getAllExternalServices(params?: PaginationParameters): Promise<PaginatedApiListResponse<ExternalService>> {
-    return this.client.get('external-services', { params })
-  }
-
-  getAllExternalServicesFromMesh({ mesh }: { mesh: string }, params?: PaginationParameters): Promise<PaginatedApiListResponse<ExternalService>> {
-    return this.client.get(`meshes/${mesh}/external-services`, { params })
-  }
-
-  getExternalService({ mesh, name }: { mesh: string, name: string }, params?: any): Promise<ExternalService> {
-    return this.client.get(`meshes/${mesh}/external-services/${name}`, { params })
-  }
-
-  getPolicyConnections({ mesh, policyType, policyName }: { mesh: string; policyType: string; policyName: string }, params?: PaginationParameters): Promise<PaginatedApiListResponse<any>> {
-    return this.client.get(`meshes/${mesh}/${policyType}/${policyName}/dataplanes`, { params })
-  }
-
-  getAllPolicyEntities({ path }: { path: string }, params?: PaginationParameters): Promise<PaginatedApiListResponse<PolicyEntity>> {
-    return this.client.get(path, { params })
-  }
-
-  getAllPolicyEntitiesFromMesh({ mesh, path }: { mesh: string, path: string }, params?: PaginationParameters): Promise<PaginatedApiListResponse<PolicyEntity>> {
-    return this.client.get(`meshes/${mesh}/${path}`, { params })
-  }
-
-  getSinglePolicyEntity({ mesh, path, name }: { mesh: string, path: string, name: string }, params?: any): Promise<PolicyEntity> {
-    return this.client.get(`meshes/${mesh}/${path}/${name}`, { params })
-  }
-}
-
-export const kumaApi = new KumaApi()
+import { get, TOKENS } from '@/services'
+export const kumaApi = get(TOKENS.api)
```

Comment:
**Question**: I noticed that it’s no longer possible to jump to the kumaApi definition from a reference. For example, in my editor, when I Alt+Click on `kumaApi` in a piece of code, it used to take me to its implementation. This is now the definition in this file instead of the actual implementation of `kumaApi`.

Looking at this further, this seems to be the case for all services/utilities accessed via the dependency inversion setup because all of their definitions are now subject to the indirection of the dependency inversion pattern (i.e. where their respective service instances are retrieved).

This is quite a downgrade in terms of my developer experience. Is there anything we can do to fix that?

Reply:
Not sure if I follow, you want to jump to the definition of `kumaApi`?

The following is with VSCode (but its the same thing in vim etc) but I'm switching 'Alt-Click' for 'Command-Click' - which I think is the same thing in my world:

1. Open up say 'ServiceDetailView.vue', goto line 103 where it says `kumaApi`, I Command-Click on that (Goto Definition) and it takes me to `@/app/kumaApi.ts`. I've moved the definition of kumaAPI into a separate file now, so that's why you don't see the class here.
2. I Command-Click on the `api` part of `TOKENS.api` in `@/app/kumaApi.ts` and it takes me to the definition of `TOKENS.api` on line 24. This shows me that `TOKENS.api` is a `KumaAPI` service.
```javascript
api: service(KumaApi, { description: 'api' }),`
```
4. I Command-Click on `KumaApi` and it takes me to `@/services/kuma-api/KumaApi.ts` (what used to be in `@/app/kumaApi.ts`

Is this a downgrade? Or did I misunderstand? I think it's maybe one click/hop more, but to be fair the class probably should have been in a separate file anyway.

Also, probably worth noting that keeping `@/api/kumaApi.ts` is just a temporary workaround as I don't want to go through the entire codebase right now changing all the `import { kumaAPI } from '@/api/kumaApi'`s to `import { useAPI } from '@/services'`

Reply:
Yes, it’s precisely what you outline in those steps: it now takes me 4 steps instead of 1 and I need to remember that the token definition will actually have a reference to the actual API wrapper class. That is a downgrade.

It just would’ve been nice if that were simpler.

Reply:
Just so I undertstand correctly, its not that you can't find the definition, its because you have to click one more time than you did previously?